### PR TITLE
Improve support for Boto 3

### PR DESCRIPTION
**Pull request by Jamim**
Hello @yurishkuro,

This PR adds support for `Boto 3` clients including enhanced support for the `S3` client.

These changes also:
 - replace `amazon/dynamodb-local` docker image with `localstack/localstack` docker image for tests
 - fix test for `SQLAlchemy`

Related issue: #101

### CODE REVIEW

**Comment by Jamim** [comment]
Hello @yurishkuro,
Could you please review this PR?
Thank you!

File: tests/opentracing_instrumentation/test_sqlalchemy.py [hunk 1]
```diff
@@ -42,9 +42,10 @@ def test_db(tracer, session):
     session.commit()
 
     spans = tracer.recorder.get_spans()
-    assert len(spans) == 3
+    assert len(spans) == 4
```
Comment (yurishkuro) [inline tests/opentracing_instrumentation/test_sqlalchemy.py]: how is this changing without any code changes?

File: .travis.yml [hunk 2]
```diff
@@ -23,16 +23,15 @@ services:
...
 before_script:
-  - docker-compose up -d dynamodb
+  - docker-compose up -d localstack
   - mysql -e 'CREATE DATABASE test;'
   - psql -c 'CREATE DATABASE test;' -U postgres
 
-  # waiting for DynamoDB
-  - while [[ $(curl -so /dev/null -w '%{response_code}' localhost:8000) != '400' ]]; do sleep 1; done
+  # waiting for the DynamoDB mock
+  - while [[ $(curl -so /dev/null -w '%{response_code}' localhost:4569) != '502' ]]; do sleep 1; done
```
Comment (yurishkuro) [inline .travis.yml]: nit: for better visibility, `do sleep 1; done` => `do echo 'waiting 1sec for DynamoDB...'; sleep 1; done`

File: .travis.yml [hunk 3]
```diff
@@ -23,16 +23,15 @@ services:
...
 
-  # workaround for Travis CI issue #7940
-  # https://github.com/travis-ci/travis-ci/issues/7940
-  - sudo rm -f /etc/boto.cfg
+  # waiting for the S3 mock
+  - while [[ $(curl -so /dev/null -w '%{response_code}' localhost:4572) != '200' ]]; do sleep 1; done
```
Comment (yurishkuro) [inline .travis.yml]: same

**Comment by yurishkuro** [comment]
> Related issue: #101

Is this fixing it?  If so, please update description to `Resolves #101`.

File: tests/opentracing_instrumentation/test_sqlalchemy.py (hunk 1 above)
Comment (Jamim) [reply tests/opentracing_instrumentation/test_sqlalchemy.py]: [Changes in SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy/commit/be985bd41ecd0eeff90c1a817167a34eb02aeed1#diff-79c290956ad49825a085d3073170ca03) cause performing one additional `PRAGMA` operation which leads to creation of a span.
That changes [are part](https://github.com/sqlalchemy/sqlalchemy/compare/rel_1_3_6...rel_1_3_7) of [SQLAlchemy 1.3.7 release](https://pypi.org/project/SQLAlchemy/1.3.7/).

I realized that I forgot to update the requirements, so I've updated them now.

File: .travis.yml (hunk 2 above)
Comment (Jamim) [reply .travis.yml]: Done.

File: .travis.yml (hunk 3 above)
Comment (Jamim) [reply .travis.yml]: Done.

**Comment by Jamim** [comment]
Hello @yurishkuro,

I'm not quite sure that this PR resolves #101 completely since @berkowitzi was asking for some example snippets. I'm going to publish them once the updated version of this library is available.
So I propose to keep #101 open and wait for @berkowitzi to confirm that everything is fine.

Thanks!

**Comment by yurishkuro** [comment]
@Jamim are you aware of any new/recent issues in Travis for dynamodb?  The builds are hanging, e.g. https://travis-ci.org/github/uber-common/opentracing-python-instrumentation/jobs/699073221

**Pull request merged**